control-plane-operator/controllers/hostedcontrolplane/ingressoperator/ingressoperator.go

```diff
@@ -118,6 +121,18 @@ func ReconcileDeployment(dep *appsv1.Deployment, params Params, apiPort *int32)
 			{Name: "IMAGE", Value: params.HAProxyRouterImage},
 			{Name: "CANARY_IMAGE", Value: params.IngressOperatorImage},
 			{Name: "KUBECONFIG", Value: "/etc/kubernetes/kubeconfig"},
+			{
+				Name:  "HTTP_PROXY",
```

Comment:
/hold

This breaks outbound connectivity to cloud provider apis which we need for public clouds.

I don't have a solution for this off-hand, but it is a problem we will have to solve for other components in the context of management cluster with proxy support as well.

Reply:
Actually it works eventually, probably after the geust cluster has nodes? But this completely negates the advantage of running it in the mgtm cluster

Reply:
Not sure how it negates it?

Reply:
You still get the benefit of it rolling out the resources ahead of time since it can talk to the kube-apis? Which is where I believe the real time benefit is: this just solves a fundamental gap that needs to be solved for health checking on the domain route.

Reply:
@alvaroaleman brought up a great point that this does call cloud provider APIs: therefore those routes should not be proxied this is added on the latest version.